libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -800,6 +800,18 @@ void CuptiActivityProfiler::finalizeTrace(const Config& config, ActivityLogger&
     logger.handleResourceInfo(resource, captureWindowStartTime_);
   }
 
+  for (auto &session : sessions_){
+    auto device_info = session->getDeviceInfo();
+    if (device_info != nullptr){
```

Comment:
nit
`if((auto device_info = session->getDeviceInfo())) {`

Reply:
if statement with initializer is a C++17 feature, if I am not mistaken Kineto currently uses C++14:

libkineto/CMakeLists.txt:
...
set_target_properties(kineto_base kineto_api PROPERTIES
CXX_STANDARD 14
CXX_STANDARD_REQUIRED YES
CXX_EXTENSIONS NO)

set(KINETO_COMPILE_OPTIONS "-DKINETO_NAMESPACE=libkineto")
list(APPEND KINETO_COMPILE_OPTIONS "-DFMT_HEADER_ONLY")
if(NOT MSVC)
list(APPEND KINETO_COMPILE_OPTIONS "-std=c++14")
else()
list(APPEND KINETO_COMPILE_OPTIONS "/std:c++14")
list(APPEND KINETO_COMPILE_OPTIONS "-DWIN32_LEAN_AND_MEAN")
list(APPEND KINETO_COMPILE_OPTIONS "-DNOGDI")
endif()
...

Reply:
ah sorry, wasn't aware of that